presto-native-execution/presto_cpp/presto_protocol/connector/clp/ClpConnectorProtocol.cpp

```
@@ -0,0 +1,167 @@
+//   typeOrdinal     : writeInt (0 = ARCHIVE, 1 = IR)
+//   kqlQueryPresent : writeBoolean
+//   [kqlQuery]      : CodecUtils.writeUtf8String, if present
+
+void ClpConnectorProtocol::deserialize(
+    const std::string& binaryData,
+    std::shared_ptr<ConnectorSplit>& proto) const {
+  std::istringstream in(binaryData);
+  auto handle = std::make_shared<ClpSplit>();
+
+  handle->path = readUTF(in);
+  int typeOrdinal = readInt(in);
```

Clang (14.0.6)

[warning] variable 'typeOrdinal' is not initialized

(cppcoreguidelines-init-variables)